cloud/linode/cilium_loadbalancers.go

```diff
@@ -443,16 +441,25 @@ func (l *loadbalancers) ensureCiliumBGPPeeringPolicy(ctx context.Context) error
 	}
 
 	// otherwise create it
-	kv := strings.Split(Options.BGPNodeSelector, "=")
-	if len(kv) != 2 {
-		return fmt.Errorf("invalid node selector %s", Options.BGPNodeSelector)
+	var nodeSelector slimv1.LabelSelector
+	// If no BGPNodeSelector is specified, select all nodes by default.
+	if Options.BGPNodeSelector == "" {
+		nodeSelector = slimv1.LabelSelector{}
+	} else {
+		kv := strings.Split(Options.BGPNodeSelector, "=")
+		if len(kv) != 2 {
+			return fmt.Errorf("invalid node selector %s", Options.BGPNodeSelector)
+		}
+
+		nodeSelector = slimv1.LabelSelector{MatchLabels: map[string]string{kv[0]: kv[1]}}
 	}
+
 	ciliumBGPPeeringPolicy := &v2alpha1.CiliumBGPPeeringPolicy{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: ciliumBGPPeeringPolicyName,
 		},
 		Spec: v2alpha1.CiliumBGPPeeringPolicySpec{
-			NodeSelector: &slimv1.LabelSelector{MatchLabels: map[string]string{kv[0]: kv[1]}},
+			NodeSelector: &nodeSelector,
```

Comment:
We should probably not have this match control plane nodes and select worker nodes only if no selector is defined